scripts/import_libcbor.py

```diff
@@ -0,0 +1,64 @@
+import os
+import tempfile
+import shutil
+import subprocess
+import re
+
+# Create a temporary directory for cloning the repository
+temp_repo_dir = tempfile.mkdtemp()
+
+try:
+    # Clone the repository into the temporary directory
+    repo_url = "https://github.com/PJK/libcbor.git"
+    clone_command = f"git clone {repo_url} {temp_repo_dir}"
+    subprocess.run(clone_command, shell=True, check=True)
+
+    # Create a separate folder for the copied files
+    output_dir = os.path.join(
+        os.path.dirname(__file__), "..", "source", "external", "libcbor")
+    os.makedirs(output_dir, exist_ok=True)
+
+    # Copy files ending with .c and .h from the /src directory
+    src_dir = os.path.join(temp_repo_dir, "src")
+
+    # Use our customized configurations
+    src_customize_cbor_dir = os.path.join(
```

Comment:
debatable: I'm not really sure if this is better or not ... but instead of additional external files, you can just have strings at the top of this file like:
```
CBOR_EXPORT_H = """
#ifndef CBOR_EXPORT_H
#define CBOR_EXPORT_H
...
"""
```

Or ... just don't bother copying these files in from anywhere, just have them live in the repo and the update script doesn't touch them.